src/core/ext/xds/xds_http_filters.cc

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <grpc/support/port_platform.h>
+
+#include "src/core/ext/xds/xds_http_filters.h"
+
+#include "envoy/extensions/filters/http/router/v3/router.upbdefs.h"
+
+namespace grpc_core {
+
+const char* kXdsHttpRouterFilterConfigName =
+    "extensions.filters.http.router.v3.Router";
+
+namespace {
+
+class XdsHttpRouterFilter : public XdsHttpFilterImpl {
+ public:
+  absl::string_view config_proto_type_name() const override {
+    return kXdsHttpRouterFilterConfigName;
+  }
+
+  void PopulateSymtab(upb_symtab* symtab) const override {
+    envoy_extensions_filters_http_router_v3_Router_getmsgdef(symtab);
+  }
+
+  absl::StatusOr<FilterConfig> GenerateFilterConfig(
+      upb_strview serialized_xds_config, upb_arena* arena) const override {
+    return FilterConfig{kXdsHttpRouterFilterConfigName, Json()};
+  }
+
+  absl::StatusOr<FilterConfig> GenerateFilterConfigOverride(
+      upb_strview serialized_xds_config, upb_arena* arena) const override {
```

Comment:
What difference will ConfigOverride has against normal Config? If we always completely override the config, then they are the same.

Reply:
It's up to individual filters to decide how to handle this.  A filter could completely replace its top-level config with the per-{VirtualHost,Route,ClusterWeight} config, or it could do the equivalent of a protobuf merge operation, or it could just selectively look at individual fields.  Some filters actually use a different protobuf message type for their top-level config and their per-{VirtualHost,Route,ClusterWeight} overrides, so there isn't any guarantee that the two configs have the same fields.